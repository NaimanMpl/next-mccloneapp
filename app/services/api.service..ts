import { RegisterFormData } from "../hooks/useRegisterForm";

export const registerUser = async (formData: RegisterFormData) => {
  const request = await fetch('/api/users', {
    method: 'POST',
    headers: { 'Content-Type' : 'application/json' },
    body: JSON.stringify(formData)
  });

  const response = await request.json();

  if (!response.ok) {
    throw new Error(response.error);
  }

  return response;
}