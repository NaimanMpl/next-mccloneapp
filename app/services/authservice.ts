import { LoginFormData } from "../hooks/useLoginForm";
import { RegisterFormData } from "../hooks/useRegisterForm";
import { UserPayload } from "../models/user.model";

export const registerUser = async (formData: RegisterFormData) => {
  const request = await fetch(
    '/api/auth/register', 
    {
      method: 'POST',
      headers: { 'Content-Type' : 'application/json' },
      body: JSON.stringify(formData)
    }
  );

  const response = await request.json();
  console.log(response)
  if (!response.ok) {
    throw new Error(response.message);
  }

  return response;
}

export const loginUser = async (formData: LoginFormData) => {
  const request = await fetch(
    '/api/auth/login', 
    {
      method: 'POST',
      headers: { 'Content-Type' : 'application/json' },
      body: JSON.stringify(formData)
    }
  );
  const response = await request.json();

  if (!response.ok) {
    throw new Error(response.message);
  }

  return response;
}

export const getCurrentUser = async (): Promise<UserPayload | null> => {
  const res = await fetch('http://localhost:3000/api/auth/me', {
    method: 'GET',
  });

  if (!res.ok) {
    return null;
  }

  return res.json();
}