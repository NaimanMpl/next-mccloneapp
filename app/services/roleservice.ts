import { RoleData } from "../models/role.model";

export const getRolesAndTheirUsers = async (): Promise<RoleData[]> => {
  const res = await fetch(
    '/api/roles/',
    {
      method: 'GET'
    }
  );

  if (!res.ok) {
    throw new Error();
  }

  return await res.json();;
}