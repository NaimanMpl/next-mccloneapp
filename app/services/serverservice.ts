import axios from 'axios';
import { Server } from '../models/server.model';

export const getServerInfo = async (): Promise<Server> => {
  try {
    const { data } = await axios.get<Server>(`/api/servers?primary=true`);
    return data;
  } catch (e) {
    return {
      id: 1,
      ip: '0.0.0.0',
      port: 50000,
      primary: true,
      lastUpdate: new Date().toString(),
      onlineSince: new Date().toString(),
      token: '',
      chatMessages: [],
      onlinePlayers: 0,
      status: 'OFFLINE',
    };
  }
};
