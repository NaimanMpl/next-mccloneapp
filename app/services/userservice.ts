import { Users } from "@prisma/client";

export const uploadSkin = async (formData: FormData, filename: string) => {
  const res = await fetch(
    '/api/user/skin/upload', 
    {
      method: 'POST',
      body: formData.get('file') as File
    }
  );
  const data = await res.json();
  if (!res.ok) {
    throw new Error(data.message);
  }
  return data;
}

export const getUsers = async (): Promise<Users[]> => {
  const res = await fetch(
    '/api/users',
    {
      method: 'GET'
    }
  );
  if (!res.ok) {
    return [];
  }

  const users = await res.json();
  return users;
}