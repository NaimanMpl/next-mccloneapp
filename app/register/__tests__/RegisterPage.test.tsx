import mockAxios from "@/__mocks__/axios";
import { render, screen, waitFor } from "@testing-library/react";
import user from '@testing-library/user-event';
import { isEmailAvailable, isUsernameAvailable } from '../../services/authservice';
import RegisterPage from "../page";

jest.mock('axios');
jest.mock('../../services/authservice');

describe('<RegisterPage />', () => {

  it('should render successfully', () => {
    const registerPage = render(
      <RegisterPage />
    );

    expect(registerPage).toMatchSnapshot();
  });

  /*
  it('should shows an error message if API call fails', async () => {

    mockAxios.post.mockRejectedValueOnce(new Error('Failed to fetch API !'));
    (isEmailAvailable as jest.Mock).mockResolvedValueOnce(true);
    (isUsernameAvailable as jest.Mock).mockResolvedValueOnce(true);

    render(
      <RegisterPage />
    );

    const [ usernameInput, emailInput ] = screen.getAllByRole('textbox');
    const [ passwordInput, confirmPasswordInput ] = screen.getAllByPlaceholderText('•••••••••••••');

    await user.clear(usernameInput);
    await user.clear(emailInput);
    await user.clear(passwordInput);
    await user.clear(confirmPasswordInput);

    await user.type(usernameInput, 'John Doe');
    await user.type(emailInput, 'johndoe@domain.com');
    await user.type(passwordInput, 'password');
    await user.type(confirmPasswordInput, 'password');

    const registerBtn = screen.getByRole('button');
    await user.click(registerBtn);

    await waitFor(() => {
      expect(screen.getByText('Le serveur a rencontré un problème.')).toBeInTheDocument();
    });

  });
  */

});