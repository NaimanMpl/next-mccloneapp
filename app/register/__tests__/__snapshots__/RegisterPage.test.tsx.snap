// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> should render successfully 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen flex items-center justify-center"
      >
        <div
          class="rounded-lg border border-border bg-card text-card-foreground shadow-sm mx-auto max-w-md"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="font-semibold tracking-tight text-2xl"
            >
              S'inscrire
            </h3>
            <p
              class="text-sm text-red-500"
            />
            <p
              class="text-sm text-muted-foreground"
            >
              Entrez vos informations pour créer un compte qui vous identifiera en jeu.
            </p>
          </div>
          <div
            class="p-6 pt-0"
          >
            <form
              class="flex flex-col gap-4"
            >
              <div
                class="space-y-2 flex flex-col"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for="username"
                >
                  Nom d'utilisateur
                </label>
                <input
                  aria-describedby=":r0:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  id=":r0:-form-item"
                  name="username"
                  placeholder="john.doe"
                  value=""
                />
              </div>
              <div
                class="space-y-2 flex flex-col"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for="email"
                >
                  Adresse mail
                </label>
                <input
                  aria-describedby=":r1:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  id=":r1:-form-item"
                  name="email"
                  placeholder="john.doe@domain.com"
                  value=""
                />
              </div>
              <div
                class="space-y-2 flex flex-col"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for="password"
                >
                  Mot de passe
                </label>
                <input
                  aria-describedby=":r2:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  id=":r2:-form-item"
                  name="password"
                  placeholder="•••••••••••••"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="space-y-2 flex flex-col"
              >
                <label
                  class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                  for="confirmPassword"
                >
                  Confirmation de mot de passe
                </label>
                <input
                  aria-describedby=":r3:-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                  id=":r3:-form-item"
                  name="confirmPassword"
                  placeholder="•••••••••••••"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-2"
              >
                <button
                  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-full"
                  type="submit"
                >
                  S'inscrire
                </button>
                <div
                  class="mt-4 text-center text-sm"
                >
                  Déjà un compte ? 
                   
                  <a
                    class="underline"
                    href="/login"
                  >
                    Se connecter
                  </a>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen flex items-center justify-center"
    >
      <div
        class="rounded-lg border border-border bg-card text-card-foreground shadow-sm mx-auto max-w-md"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="font-semibold tracking-tight text-2xl"
          >
            S'inscrire
          </h3>
          <p
            class="text-sm text-red-500"
          />
          <p
            class="text-sm text-muted-foreground"
          >
            Entrez vos informations pour créer un compte qui vous identifiera en jeu.
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <form
            class="flex flex-col gap-4"
          >
            <div
              class="space-y-2 flex flex-col"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                for="username"
              >
                Nom d'utilisateur
              </label>
              <input
                aria-describedby=":r0:-form-item-description"
                aria-invalid="false"
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                id=":r0:-form-item"
                name="username"
                placeholder="john.doe"
                value=""
              />
            </div>
            <div
              class="space-y-2 flex flex-col"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                for="email"
              >
                Adresse mail
              </label>
              <input
                aria-describedby=":r1:-form-item-description"
                aria-invalid="false"
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                id=":r1:-form-item"
                name="email"
                placeholder="john.doe@domain.com"
                value=""
              />
            </div>
            <div
              class="space-y-2 flex flex-col"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                for="password"
              >
                Mot de passe
              </label>
              <input
                aria-describedby=":r2:-form-item-description"
                aria-invalid="false"
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                id=":r2:-form-item"
                name="password"
                placeholder="•••••••••••••"
                type="password"
                value=""
              />
            </div>
            <div
              class="space-y-2 flex flex-col"
            >
              <label
                class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
                for="confirmPassword"
              >
                Confirmation de mot de passe
              </label>
              <input
                aria-describedby=":r3:-form-item-description"
                aria-invalid="false"
                class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
                id=":r3:-form-item"
                name="confirmPassword"
                placeholder="•••••••••••••"
                type="password"
                value=""
              />
            </div>
            <div
              class="flex flex-col gap-2"
            >
              <button
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-semibold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-full"
                type="submit"
              >
                S'inscrire
              </button>
              <div
                class="mt-4 text-center text-sm"
              >
                Déjà un compte ? 
                 
                <a
                  class="underline"
                  href="/login"
                >
                  Se connecter
                </a>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
